components/script/dom/css.rs

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::error::{Error, Fallible};
+use dom::bindings::utils::Reflector;
+use util::str::DOMString;
+
+use cssparser::serialize_identifier;
+
+#[dom_struct]
+pub struct CSS {
+    reflector_: Reflector,
+}
+
+impl CSS {
+    // http://dev.w3.org/csswg/cssom/#serialize-an-identifier
+    pub fn Escape(ident: DOMString) -> Fallible<DOMString> {
+        if ident.chars().any(|c| c == '\x00') {
```

Comment:
This can be replaced by `ident.bytes().any(|b| b == b'\0')`, which is equivalent but does less work. (`chars` does UTF-8 decoding.)